### PR TITLE
Use the Java 8 Date/Time API.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
- [x] I carefully read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md) and agree to them.
- [x] I have tested the API against [NewPipe](https://github.com/TeamNewPipe/NewPipe).
- [x] I agree to create a pull request for [NewPipe](https://github.com/TeamNewPipe/NewPipe) as soon as possible to make it compatible with the changed API.

### CODE REVIEW

**Comment by wb9688** [comment]
D8/R8 supports desugaring up to Java 12. (However AGP only supports Java 8 due to `bootclasspath`, so it's limited to Java 8 for Android code i.e. NewPipe, while you could use Java 12 in non-Android code i.e. NewPipeExtractor.)

**Comment by Isira-Seneviratne** [comment]
> D8/R8 supports desugaring up to Java 12. (However AGP only supports Java 8 due to `bootclasspath`, so it's limited to Java 8 for Android code i.e. NewPipe, while you could use Java 12 in non-Android code i.e. NewPipeExtractor.)

So should I set the JDK version to 12 instead of 8?

**Comment by Isira-Seneviratne** [comment]
The test failures appear to be due to outdated links being used.

**Review by Stypox — APPROVED** [review]
Looks good to me, I haven't yet tested though

**Review by TobiGr — APPROVED** [review]

**Comment by TobiGr** [comment]
Ah. Well. That was not a good review :laughing: We missed, that the PR does not even build and breaks some tests...

File: extractor/src/main/java/org/schabi/newpipe/extractor/localization/DateWrapper.java [hunk 1]
```diff
@@ -3,30 +3,61 @@
...
+        this(calendar, false);
+    }
+
+    /**
+     * @deprecated Use {@link #DateWrapper(OffsetDateTime, boolean)} instead.
+     */
+    @Deprecated
+    public DateWrapper(@NonNull Calendar calendar, boolean isApproximation) {
+        offsetDateTime = OffsetDateTime.ofInstant(calendar.toInstant(), ZoneOffset.UTC);
+        this.isApproximation = isApproximation;
```
Comment (B0pol) [inline extractor/src/main/java/org/schabi/newpipe/extractor/localization/DateWrapper.java:32]: you should use 
```java
this(OffsetDateTime.ofInstant(calendar.toInstant(), ZoneOffset.UTC), isApproximation);
```

Comment (Isira-Seneviratne) [reply extractor/src/main/java/org/schabi/newpipe/extractor/localization/DateWrapper.java:32]: Shall I commit this in a separate PR?

Comment (B0pol) [reply extractor/src/main/java/org/schabi/newpipe/extractor/localization/DateWrapper.java:32]: That would be ideal, thanks

**Pull request merged**